File: packages/python/src/openhook/events.py
"""Standard OpenHook event types."""

from enum import StrEnum


class EventType(StrEnum):
    SESSION_START = "session.start"
    SESSION_END = "session.end"
    PROMPT_SUBMIT = "prompt.submit"
    TOOL_START = "tool.start"
    TOOL_END = "tool.end"
    FILE_WRITE = "file.write"


ALL_EVENT_TYPES = frozenset(EventType)

File: packages/typescript/src/events.ts
export const EventType = {
  SessionStart: "session.start",
  SessionEnd: "session.end",
  PromptSubmit: "prompt.submit",
  ToolStart: "tool.start",
  ToolEnd: "tool.end",
  FileWrite: "file.write",
} as const;

export type EventTypeValue = (typeof EventType)[keyof typeof EventType];

const ALL_EVENT_TYPES = new Set<string>(Object.values(EventType));

export function isValidEventType(value: string): value is EventTypeValue {
  return ALL_EVENT_TYPES.has(value);
}
